core/arch/arm/include/kernel/early_ta.h

```diff
@@ -19,5 +21,11 @@ struct early_ta {
 
 #define __early_ta __section(".rodata.early_ta" __SECTION_FLAGS_RODATA)
 
+#define for_each_early_ta(_ta) \
```

Comment:
I known that are already macros with lowercase names. But maybe prefer here `FOR_EACH_EARLY_TA()`

Reply:
This is an exception, like in the Linux kernel:
```
linux $ git grep "#define for_each" | wc -l
155
linux $ git grep "#define FOR_EACH" | wc -l
0
```
I prefer to keep it lowercase.

Reply:
```shell
$ grep -rs "#define" */ | grep "for_each" 
core/lib/libfdt/include/libfdt.h:#define fdt_for_each_subnode(node, fdt, parent)		\
core/lib/libfdt/include/libfdt.h:#define fdt_for_each_property_offset(property, fdt, node)	\
core/include/kernel/dt.h:#define for_each_dt_driver(drv) \
core/arch/arm/kernel/early_ta.c:#define for_each_early_ta(_ta) \
$
$ grep -rs "#define" */ | grep "FOREACH" 
core/include/scattered_array.h:#define SCATTERED_ARRAY_FOREACH(elem, array_name, element_type) \
lib/libutils/isoc/bget_malloc.c:#define BPOOL_FOREACH(ctx, iterator, bp)		      \
lib/libutils/isoc/include/sys/queue.h:#define	LIST_FOREACH(var, head, field)					\
lib/libutils/isoc/include/sys/queue.h:#define	SLIST_FOREACH(var, head, field)					\
lib/libutils/isoc/include/sys/queue.h:#define	STAILQ_FOREACH(var, head, field)				\
lib/libutils/isoc/include/sys/queue.h:#define	STAILQ_FOREACH_SAFE(var, head, field, tvar)			\
lib/libutils/isoc/include/sys/queue.h:#define	SIMPLEQ_FOREACH(var, head, field)				\
lib/libutils/isoc/include/sys/queue.h:#define	TAILQ_FOREACH(var, head, field)					\
lib/libutils/isoc/include/sys/queue.h:#define	TAILQ_FOREACH_SAFE(var, head, field, next)			\
lib/libutils/isoc/include/sys/queue.h:#define	TAILQ_FOREACH_REVERSE(var, head, headname, field)		\
lib/libutils/isoc/include/sys/queue.h:#define	TAILQ_FOREACH_REVERSE_SAFE(var, head, headname, field, prev)	\
lib/libutils/isoc/include/sys/queue.h:#define	CIRCLEQ_FOREACH(var, head, field)				\
lib/libutils/isoc/include/sys/queue.h:#define	CIRCLEQ_FOREACH_REVERSE(var, head, field)			\
out/arm/export-ta_arm32/include/sys/queue.h:#define	LIST_FOREACH(var, head, field)					\
out/arm/export-ta_arm32/include/sys/queue.h:#define	SLIST_FOREACH(var, head, field)					\
out/arm/export-ta_arm32/include/sys/queue.h:#define	STAILQ_FOREACH(var, head, field)				\
out/arm/export-ta_arm32/include/sys/queue.h:#define	STAILQ_FOREACH_SAFE(var, head, field, tvar)			\
out/arm/export-ta_arm32/include/sys/queue.h:#define	SIMPLEQ_FOREACH(var, head, field)				\
out/arm/export-ta_arm32/include/sys/queue.h:#define	TAILQ_FOREACH(var, head, field)					\
out/arm/export-ta_arm32/include/sys/queue.h:#define	TAILQ_FOREACH_SAFE(var, head, field, next)			\
out/arm/export-ta_arm32/include/sys/queue.h:#define	TAILQ_FOREACH_REVERSE(var, head, headname, field)		\
out/arm/export-ta_arm32/include/sys/queue.h:#define	TAILQ_FOREACH_REVERSE_SAFE(var, head, headname, field, prev)	\
out/arm/export-ta_arm32/include/sys/queue.h:#define	CIRCLEQ_FOREACH(var, head, field)				\
out/arm/export-ta_arm32/include/sys/queue.h:#define	CIRCLEQ_FOREACH_REVERSE(var, head, field)			\

Reply:
Linux vs. BSD fight ;-)

Reply:
I have no strong opinion but i prefer macros in uppercase...
I let you @jenswi-linaro and @jbech-linaro decide on this.

Reply:
Same here, no strong opinion. I'm generally for upper case when it's about define's, but if we're "copying" things from other locations, then I also see a point keeping it like it's at the source. But if no-one are willing to make a decision, then I'd say go with upper case.

Reply:
I'd rather not change it, we're just moving it after all. If we're about to change this, please take it in another PR (and have the discussion there) to let this PR keep the focus.